### PR TITLE
Update tutorial.rst to include section about accessing Zip Files on S3

### PR DESCRIPTION
**Pull request by jeffpeck10x**
Per discussion here, add information about about accessing zip files on s3: https://github.com/zarr-developers/zarr-python/discussions/1613

[Description of PR]

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
Thanks, @jeffpeck10x! I've triggered the checks.

**Comment by joshmoore** [comment]
Beautiful!

image


Would you like to add yourself to [docs/release.txt](https://github.com/zarr-developers/zarr-python/blob/main/docs/release.rst)

**Review by joshmoore — APPROVED** [review]

**Comment by jeffpeck10x** [comment]
Thank you @joshmoore . I have updated the release notes.

File: docs/tutorial.rst [hunk 1]
```diff
@@ -1000,6 +1000,32 @@ separately from Zarr.
 
 .. _tutorial_copy:
 
+Accessing Zip Files on S3
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The built-in `ZipStore` will only work with paths on the local file-system, however
+it is also possible to access ``.zarr.zip`` data on the cloud. Here is an example of
```
Comment (d-v-b) [inline docs/tutorial.rst:1007]: I worry that this framing implies that there is something special about the ``.zarr.zip`` string at the end of a filename.  Could we adjust the language to  say "zipped zarr" or something that doesn't assume a particular filename convention?

File: docs/tutorial.rst [hunk 2]
```diff
@@ -1000,6 +1000,32 @@ separately from Zarr.
...
+    >>> 
+    >>> # cache is optional, but may be a good idea depending on the situation
+    >>> cache = zarr.storage.LRUStoreCache(store, max_size=2**28)
+    >>> z = zarr.group(store=cache)
+
+This store can also be generated with ``fsspec``'s handler chaining, like so:
+
+    >>> store = zarr.storage.FSStore(url=f"zip::{s3_path}",  mode="r")
+
+Note that this is intended for a read-only data source. However, this can be
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/tutorial.rst]: I've pushed a commit to remove that wording, @d-v-b. https://github.com/zarr-developers/outreachy_2022_testing_zipstore/tree/main certainly lists fsspec/write as supported.

File: docs/tutorial.rst (hunk 1 above)
Comment (joshmoore) [reply docs/tutorial.rst:1007]: Pushed.

**Comment by joshmoore** [comment]
Thanks for the review, @d-v-b, and @jeffpeck10x, for getting this started! :tada: If anyone else would like to further improve these (or other :smile:) docs, please feel free.

**Comment by d-v-b** [comment]
Sorry for being a pedant, but could we revert? I'm not done with my suggestions :) For example,  the new docs still contain references to `zarr.zip` where I think we should be saying 'zipped zarr hierarchy' or similar

File: docs/tutorial.rst [hunk 3]
```diff
@@ -1000,6 +1000,31 @@ separately from Zarr.
 
 .. _tutorial_copy:
 
+Accessing Zip Files on S3
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The built-in `ZipStore` will only work with paths on the local file-system, however
```
Comment (d-v-b) [inline docs/tutorial.rst:1006]: I'm not sure how links work in sphinx, but could we make this a link to the API docs for `ZipStore`?

File: docs/tutorial.rst [hunk 4]
```diff
@@ -1000,6 +1000,31 @@ separately from Zarr.
...
+it is also possible to access ``.zarr.zip`` data on the cloud. Here is an example of
+accessing a zipped Zarr file on s3:
```
Comment (d-v-b) [inline docs/tutorial.rst:1008]: I think I prefer "zipped Zarr hierarchy" to "zipped Zarr file"

File: docs/tutorial.rst [hunk 5]
```diff
@@ -1000,6 +1000,31 @@ separately from Zarr.
...
+    >>> 
+    >>> # cache is optional, but may be a good idea depending on the situation
+    >>> cache = zarr.storage.LRUStoreCache(store, max_size=2**28)
+    >>> z = zarr.group(store=cache)
+
+This store can also be generated with ``fsspec``'s handler chaining, like so:
+
+    >>> store = zarr.storage.FSStore(url=f"zip::{s3_path}",  mode="r")
+
+This can be especially useful if you have a very large ``.zarr.zip`` file on s3
```
Comment (d-v-b) [inline docs/tutorial.rst:1025]: same as above -- let's replace `zarr.zip` with "zipped Zarr hierarchy"

File: docs/tutorial.rst [hunk 6]
```diff
@@ -1000,6 +1000,31 @@ separately from Zarr.
...
+accessing a zipped Zarr file on s3:
+
+    >>> s3_path = "s3://path/to/my.zarr.zip"
+    >>> 
+    >>> s3 = s3fs.S3FileSystem()
+    >>> f = s3.open(s3_path)
+    >>> fs = ZipFileSystem(f, mode="r")
+    >>> store = FSMap("", fs, check=False)
+    >>> 
+    >>> # cache is optional, but may be a good idea depending on the situation
```
Comment (d-v-b) [inline docs/tutorial.rst:1017]: Which situations benefit from the cache?

**Comment by d-v-b** [comment]
I'm also happy to make the changes myself, if people broadly agree with what I'm proposing

**Comment by d-v-b** [comment]
@joshmoore any objections if I revert the merge and make a few tweaks, then re-merge?

**Comment by joshmoore** [comment]
Let's just start with a new PR from your side. Then @jeffpeck10x can decide if he's going to jump back in (rather than *definitely* getting the notifications, etc. 🙂)

**Pull request merged**